src/Data/PQueue/Prio/Internals.hs

```diff
@@ -709,6 +727,16 @@ traverseWithKeyU :: Applicative f => (k -> a -> f b) -> MinPQueue k a -> f (MinP
 traverseWithKeyU _ Empty = pure Empty
 traverseWithKeyU f (MinPQ n k a ts) = liftA2 (MinPQ n k) (f k a) (traverseForest f (const (pure Zero)) ts)
 
+-- | \(O(n)\). An unordered traversal over a priority queue, in no particular order.
+-- While there is no guarantee in which order the elements are traversed, the resulting
+-- priority queue will be perfectly valid. The results are forced before they are installed
+-- in the queue.
+--
+-- @since 1.5.0
+traverseWithKeyU' :: Applicative f => (k -> a -> f b) -> MinPQueue k a -> f (MinPQueue k b)
+traverseWithKeyU' _ Empty = pure Empty
+traverseWithKeyU' f (MinPQ n k a ts) = liftA2 (\ !b !q' -> MinPQ n k b q') (f k a) (traverseForest' f (const (pure Zero)) ts)
```

Comment:
Forcing `q'` shouldn't make a difference, since the field is strict anyway.

Reply:
Good point. I admit I got lazy and didn't check the constructor.

Reply:
OK, after going through and removing the unneeded bangs, I found that it was a chore to be sure I was banging exactly the right things. In this limited context, it's totally harmless to just bang everything—GHC won't have any trouble with the extras. So I just put them back in. There are loads of places we have to be careful not to force things prematurely; this ain't one.